packages/flutter/lib/src/material/dropdown.dart

```diff
@@ -43,21 +43,23 @@ class _DropdownMenuPainter extends CustomPainter {
     this.color,
     this.elevation,
     this.selectedIndex,
+    this.borderRadius,
     required this.resize,
     required this.getSelectedItemOffset,
   }) : _painter = BoxDecoration(
          // If you add an image here, you must provide a real
          // configuration in the paint() function and you must provide some sort
          // of onChanged callback here.
          color: color,
-         borderRadius: BorderRadius.circular(2.0),
+         borderRadius: borderRadius ?? BorderRadius.circular(2.0),
```

Comment:
FWIW, you can use a const value for the default: `const BorderRadius.all(Radius.circular(2))`